config/routes.rb

```diff
@@ -18,6 +18,16 @@
 
   mount Repp::API => '/'
 
+  namespace :api do
+    namespace :v1 do
+      namespace :registrant do
+        post 'auth/eid', to: 'auth#eid'
+
+        resources :domains, only: [:index, :show], param: :uuid
```

Comment:
How about adding `uuid` param constraint?

`constraints uuid: /[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89abAB][0-9a-f]{3}-[0-9a-f]{12}/ do
end`

Reply:
I don't think there's anything to be gained from that. It technically would save a roundtrip to the database, but it's fairly uncommon to see anything in the mould of `resources :posts, id: /\d+/`, which is far cheaper an easier to maintain. By default, data is requested from the database with even the most bizarre ids. 

<img width="1018" alt="screen shot 2018-08-04 at 17 06 34" src="https://user-images.githubusercontent.com/2331765/43677318-c84eeabe-9808-11e8-9070-6132462334db.png">


Not to mention that a constraint that covers the complexity of a UUID, as you commented with is complex enough to be problematic to test.

Reply:
Database round-trip is not relevant, that's for sure. My point was to set up the router to reject invalid requests (say `registrant-api.eis.ee/domains/123456`).

Why `/\d+/`? Don't we use UUID of format x-x-x-x?

Reply:
> My point was to set up the router to reject invalid requests (say `registrant-api.eis.ee/domains/123456`).

What would be the benefit? The regex is very complex and we both agree that a database query against an index is not that painful.

If there's something to be gained from such a constraint, we should have an analogous one for other resources identified via primary keys, such as `/registrar/invoices/123`, where it would take form of `/\d+/`. We do not.